# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1283290038 -7200
#      Tue Aug 31 23:27:18 2010 +0200
# Node ID f6294203286e10f53a39a0a9550c795c5bcd2ea7
# Parent  e78e531dfa2db306e7c24ff0996dc51c5d4d8d53
Fix off by one error in the what function (bug #30919)

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,12 @@
 2010-07-07  David Bateman  <dbateman@free.fr>
 
+	* miscellaneous/what.m: Off by one error.
+
 	* plot/__go_draw_axes__.m: Improve matlab compatibility for "."
 	marker.
 
 2010-08-31  Ben Abbott <bpabbott@mac.com>
 
 	* plot/private/__actual_axis_position__.m, plot/__go_draw_axes__.m,
 	plot/axis.m, plot/colorbar.m, plot/daspect.m, plot/plotyy.m,
 	plot/subplot.m: Treatment of activepositionproperty for gnuplot.
diff --git a/scripts/miscellaneous/what.m b/scripts/miscellaneous/what.m
--- a/scripts/miscellaneous/what.m
+++ b/scripts/miscellaneous/what.m
@@ -98,14 +98,14 @@ function __display_filenames__ (msg, p, 
       fmt = sprintf ("%s   %%-%ds", fmt, maxlen);
     endfor
     fmt = [fmt, "\n"];
 
     nrows = ceil (length (f) / ncols); 
     for i = 1 : nrows
       args  = f(i:nrows:end);
       if (length (args) < ncols)
-        args(end : ncols) = {""};
+        args(end + 1 : ncols) = {""};
       endif
       printf (fmt, args{:});
     endfor
   endif
 endfunction
